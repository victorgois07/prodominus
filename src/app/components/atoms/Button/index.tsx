import { ButtonHTMLAttributes, forwardRef } from "react";
import { twMerge } from "tailwind-merge";

export const Button = forwardRef<
  HTMLButtonElement,
  ButtonHTMLAttributes<HTMLButtonElement>
>(({ className, ...props }, ref) => (
  <button
    ref={ref}
    className={twMerge(
      "px-4 py-2 rounded font-semibold transition disabled:opacity-50",
      className
    )}
    {...props}
  />
));
Button.displayName = "Button";
