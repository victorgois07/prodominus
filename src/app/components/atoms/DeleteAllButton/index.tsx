"use client";

import { useProducts } from "@/business/hooks/useProducts";
import { useConfirmDialog } from "@/business/store/useConfirmDialog";

export function DeleteAllButton() {
  const { deleteAllProducts } = useProducts();
  const { show } = useConfirmDialog();

  const handleDeleteAll = () => {
    show(
      "Tem certeza que deseja excluir todos os produtos? Esta ação não pode ser desfeita.",
      async () => {
        await deleteAllProducts();
      }
    );
  };

  return (
    <button
      onClick={handleDeleteAll}
      className="bg-red-600 hover:bg-red-700 text-white px-4 py-2 rounded-md transition-colors"
    >
      Excluir Todos
    </button>
  );
}
