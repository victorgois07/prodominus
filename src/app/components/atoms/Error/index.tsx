interface ErrorProps {
  message: string;
}

export function Error({ message }: ErrorProps) {
  return (
    <div className="flex items-center justify-center min-h-[400px]">
      <div
        className="bg-red-50 border border-red-200 text-red-700 px-4 py-3 rounded relative"
        role="alert"
      >
        <strong className="font-bold">Error: </strong>
        <span className="block sm:inline">{message}</span>
      </div>
    </div>
  );
}
