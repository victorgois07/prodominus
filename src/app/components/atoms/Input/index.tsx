import { InputHTMLAttributes, forwardRef } from "react";
import { twMerge } from "tailwind-merge";

export const Input = forwardRef<
  HTMLInputElement,
  InputHTMLAttributes<HTMLInputElement>
>(({ className, ...props }, ref) => (
  <input
    ref={ref}
    className={twMerge(
      "shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline",
      className
    )}
    {...props}
  />
));
Input.displayName = "Input";
