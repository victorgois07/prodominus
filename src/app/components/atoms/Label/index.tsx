import { LabelHTMLAttributes, forwardRef } from "react";

export const Label = forwardRef<
  HTMLLabelElement,
  LabelHTMLAttributes<HTMLLabelElement>
>((props, ref) => (
  <label ref={ref} className="block mb-1 font-medium" {...props} />
));
Label.displayName = "Label";
