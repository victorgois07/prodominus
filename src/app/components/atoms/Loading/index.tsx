export function Loading() {
  return (
    <div className="flex items-center justify-center min-h-[400px]">
      <div className="animate-spin rounded-full h-12 w-12 border-t-2 border-b-2 border-green-500"></div>
    </div>
  );
}
