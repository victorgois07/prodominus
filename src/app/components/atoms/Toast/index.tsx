import { useEffect } from "react";

interface ToastProps {
  message: string;
  type: "success" | "error";
  onClose: () => void;
}

export function Toast({ message, type, onClose }: ToastProps) {
  useEffect(() => {
    const timer = setTimeout(onClose, 3000);
    return () => clearTimeout(timer);
  }, [onClose]);

  return (
    <div
      className={`fixed bottom-4 right-4 px-6 py-3 rounded-lg shadow-lg text-white ${
        type === "success" ? "bg-green-500" : "bg-red-500"
      }`}
    >
      <div className="flex items-center justify-between">
        <span>{message}</span>
        <button
          onClick={onClose}
          className="ml-4 text-white hover:text-gray-200"
          aria-label="close"
        >
          âœ•
        </button>
      </div>
    </div>
  );
}
