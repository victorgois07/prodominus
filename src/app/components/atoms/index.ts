export * from "./Button";
export * from "./Error";
export * from "./Input";
export * from "./Label";
export * from "./Loading";
export * from "./Toast";
