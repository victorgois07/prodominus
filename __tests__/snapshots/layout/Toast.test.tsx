import { Toast } from "@/app/components/atoms/Toast";
import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";

describe("Toast Snapshot", () => {
  it("should match snapshot with success message", () => {
    const { container } = render(
      <Toast
        message="Operation completed successfully!"
        type="success"
        onClose={() => {}}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with error message", () => {
    const { container } = render(
      <Toast
        message="An error occurred. Please try again."
        type="error"
        onClose={() => {}}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with long message", () => {
    const { container } = render(
      <Toast
        message="This is a very long toast message that should be properly displayed and wrapped if necessary. It contains multiple sentences and should be properly formatted."
        type="success"
        onClose={() => {}}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with short message", () => {
    const { container } = render(
      <Toast message="Done!" type="success" onClose={() => {}} />
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with error message containing special characters", () => {
    const { container } = render(
      <Toast
        message="Error: Invalid input! Please check your data (code: #123)"
        type="error"
        onClose={() => {}}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
