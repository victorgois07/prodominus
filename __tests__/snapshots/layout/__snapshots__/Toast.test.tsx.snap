// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Toast Snapshot > should match snapshot with error message 1`] = `
<div>
  <div
    class="fixed bottom-4 right-4 px-6 py-3 rounded-lg shadow-lg text-white bg-red-500"
  >
    <div
      class="flex items-center justify-between"
    >
      <span>
        An error occurred. Please try again.
      </span>
      <button
        aria-label="close"
        class="ml-4 text-white hover:text-gray-200"
      >
        ✕
      </button>
    </div>
  </div>
</div>
`;

exports[`Toast Snapshot > should match snapshot with error message containing special characters 1`] = `
<div>
  <div
    class="fixed bottom-4 right-4 px-6 py-3 rounded-lg shadow-lg text-white bg-red-500"
  >
    <div
      class="flex items-center justify-between"
    >
      <span>
        Error: Invalid input! Please check your data (code: #123)
      </span>
      <button
        aria-label="close"
        class="ml-4 text-white hover:text-gray-200"
      >
        ✕
      </button>
    </div>
  </div>
</div>
`;

exports[`Toast Snapshot > should match snapshot with long message 1`] = `
<div>
  <div
    class="fixed bottom-4 right-4 px-6 py-3 rounded-lg shadow-lg text-white bg-green-500"
  >
    <div
      class="flex items-center justify-between"
    >
      <span>
        This is a very long toast message that should be properly displayed and wrapped if necessary. It contains multiple sentences and should be properly formatted.
      </span>
      <button
        aria-label="close"
        class="ml-4 text-white hover:text-gray-200"
      >
        ✕
      </button>
    </div>
  </div>
</div>
`;

exports[`Toast Snapshot > should match snapshot with short message 1`] = `
<div>
  <div
    class="fixed bottom-4 right-4 px-6 py-3 rounded-lg shadow-lg text-white bg-green-500"
  >
    <div
      class="flex items-center justify-between"
    >
      <span>
        Done!
      </span>
      <button
        aria-label="close"
        class="ml-4 text-white hover:text-gray-200"
      >
        ✕
      </button>
    </div>
  </div>
</div>
`;

exports[`Toast Snapshot > should match snapshot with success message 1`] = `
<div>
  <div
    class="fixed bottom-4 right-4 px-6 py-3 rounded-lg shadow-lg text-white bg-green-500"
  >
    <div
      class="flex items-center justify-between"
    >
      <span>
        Operation completed successfully!
      </span>
      <button
        aria-label="close"
        class="ml-4 text-white hover:text-gray-200"
      >
        ✕
      </button>
    </div>
  </div>
</div>
`;
